lxd/storage/backend_lxd.go

```diff
@@ -4183,6 +4183,8 @@ func (b *lxdBackend) CreateCustomVolumeFromCopy(projectName string, srcProjectNa
 
 	if contentDBType == db.StoragePoolVolumeContentTypeBlock {
```

Comment:
This is starting to feel like it would be clearer as a switch statement with a default case of returning an error with an unsupported content type error.

Reply:
Why would we want to have a default case returning an error? `VolumeContentTypeNameToContentType()` will return an error if the content type is unknown. A `switch` statement wouldn't make any difference.

Reply:
In that case we should use (or create if doesnt exist) a generic function that converts between `contentDBType` and `contentType` and use it here. My general theme with these comments is to try and get away from defining exceptions and instead moving towards what is allowed and then catching anything that is outside of that as an error.

Its true that I mistakenly thought this was one of those checks, but if its just a generic type conversion this is likely going to be needed elsewhere too so would be good to get into its own function.

Reply:
Ah we already have this in the form of `VolumeDBContentTypeToContentType` I think.

Reply:
> In that case we should use (or create if doesnt exist) a generic function that converts between contentDBType and contentType and use it here. 

I just looked into this, but that specific function doesn't exist. Instead, we could do the following though:
```go
	contentDBType, err := VolumeContentTypeNameToContentType(srcConfig.Volume.ContentType)
	if err != nil {
		return err
	}

	// Get the source volume's content type.
	contentType, _ := VolumeDBContentTypeToContentType(contentDBType)
```

We wouldn't need to catch the second error because an invalid content type would error in the first function.

Reply:
Does `VolumeDBContentTypeToContentType` not do what we need then?

Reply:
> We dont need one because we've already got the contentDBType above we can use with VolumeDBContentTypeToContentType

I'm confused. That's exactly what's in my code example, so all is good, no? :)

Reply:
The code in this PR is:

```go
contentDBType, err := VolumeContentTypeNameToContentType(srcConfig.Volume.ContentType)
if err != nil {
    return err
}

// Get the source volume's content type.
contentType := drivers.ContentTypeFS

if contentDBType == db.StoragePoolVolumeContentTypeBlock {
    contentType = drivers.ContentTypeBlock
} else if contentDBType == db.StoragePoolVolumeContentTypeISO {
    contentType = drivers.ContentTypeISO
}
```

But we can use the existing `VolumeDBContentTypeToContentType` function here instead of partially re-implementing it:

```go
contentDBType, err := VolumeContentTypeNameToContentType(srcConfig.Volume.ContentType)
if err != nil {
    return err
}

contentType, err := VolumeDBContentTypeToContentType(contentDBType)
if err != nil {
    return err
}
```

Reply:
Exactly, that's in my comment above. So all's good. However, there's no need to check the second error as any content type error will be caught in the first function.

Reply:
Yes, I didn't follow what you meant. But please do check the error, we never know what may happen in the future if one function gets updated but the other doesn't or some such, for the sake of 3 lines its safer.